reporters/kamon-prometheus/src/main/scala/kamon/prometheus/embeddedhttp/SunEmbeddedHttpServer.scala

```diff
@@ -32,24 +35,40 @@ class SunEmbeddedHttpServer(hostname: String, port: Int, scrapeSource: ScrapeSou
         if (httpExchange.getRequestURI.getPath == "/metrics") {
           val data = scrapeSource.scrapeData()
           val bytes = data.getBytes(StandardCharsets.UTF_8)
-          httpExchange.sendResponseHeaders(200, bytes.length)
           val os = httpExchange.getResponseBody
           try {
             os.write(bytes)
-          }
-          finally
-            os.close()
-        }
-        else {
-          httpExchange.sendResponseHeaders(404, -1)
-        }
+            if (shouldUseCompression(httpExchange)) {
+              val gzip = new GZIPOutputStream(os)
+              httpExchange.sendResponseHeaders(200, 0)
+              gzip.write(bytes)
+              gzip.close()
+            } else httpExchange.sendResponseHeaders(200, bytes.length)
+          } finally os.close()
+        } else httpExchange.sendResponseHeaders(404, -1)
       }
-
     }
+
     s.createContext("/metrics", handler)
     s.start()
     s
   }
 
   def stop(): Unit = server.stop(0)
 }
+
+object SunEmbeddedHttpServer {
+  def shouldUseCompression(httpExchange: HttpExchange): Boolean = {
+    val encodingHeaders = httpExchange.getRequestHeaders.asScala.get("Accept-Encoding")
+    encodingHeaders match {
+      case Some(headerList) =>
+        val trimmedEncodings = for {
+          encodingHeader <- headerList.asScala
+          encodings = encodingHeader.split(",")
+          encoding <- encodings
+        } yield encoding.trim().toLowerCase()
+        trimmedEncodings.contains("gzip")
+      case None => false
+    }
+  }
+}
```

Comment:
You don't have to do this, but configuring your editor to add newlines at the end of files will stop github from showing this :D

Reply:
Added in https://github.com/kamon-io/Kamon/pull/915/commits/d98f3506d36dbfb309334a1bfe3f61622dca6dc0.